sphinx/ext/mathjax.py

```diff
@@ -105,9 +105,26 @@ def install_mathjax(
                 )
             body = 'MathJax.Hub.Config(%s)' % json.dumps(app.config.mathjax2_config)
             builder.add_js_file('', type='text/x-mathjax-config', body=body)
-        if app.config.mathjax3_config:
-            body = 'window.MathJax = %s' % json.dumps(app.config.mathjax3_config)
-            builder.add_js_file('', body=body)
+        match app.config.mathjax3_config:
+            case None:
+                pass
+            case str(config_filename):
+                config_filepath = app.srcdir / config_filename
+                if not config_filepath.exists():
+                    msg = f'mathjax3_config file not found: {config_filepath!s}'
+                    raise ExtensionError(msg)
+                if not config_filepath.is_file() or config_filepath.suffix != '.js':
```

Comment:
Are there other extension than `.js` to consider? I am not familiar with javascript